File: fix_entities.py
import os

files = [
    r'frontend\src\screens\FeeManagementScreen.js',
    r'backend\controllers\studentController.js',
    r'backend\controllers\feeController.js'
]

for file_path in files:
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Replace HTML entities
        content = content.replace('&amp;gt;', '>')
        content = content.replace('&amp;lt;', '<')
        content = content.replace('&amp;amp;', '&')
        content = content.replace('&gt;', '>')
        content = content.replace('&lt;', '<')
        content = content.replace('&amp;', '&')
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        print(f'✓ Fixed {file_path}')
    except Exception as e:
        print(f'✗ Error fixing {file_path}: {e}')

print('\nAll files fixed!')

File: fix-entities.js
const fs = require('fs');
const path = require('path');

const files = [
    'frontend/src/screens/FeeManagementScreen.js',
    'backend/controllers/studentController.js',
    'backend/controllers/feeController.js'
];

files.forEach(file => {
    const filePath = path.join(__dirname, file);
    try {
        let content = fs.readFileSync(filePath, 'utf8');
        content = content.replace(/&gt;/g, '>');
        content = content.replace(/&lt;/g, '<');
        content = content.replace(/&amp;/g, '&');
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`✓ Fixed ${file}`);
    } catch (err) {
        console.error(`✗ Error fixing ${file}:`, err.message);
    }
});

console.log('\nDone!');
